source/bundling-a-java-app.html.haml.markdown

```diff
@@ -62,14 +70,94 @@
+  &lt;component type="desktop"&gt;
+   &lt;id&gt;com.sweethome3d.App.desktop&lt;/id&gt;
+   &lt;metadata_license&gt;CC0-1.0&lt;/metadata_license&gt;
+   &lt;project_license&gt;GPL-2.0&lt;/project_license&gt;
+   &lt;name&gt;SweetHome 3D&lt;/name&gt;
+   &lt;summary&gt;Draw the plan of your home or office, test furniture layouts and view the results in 3D&lt;/summary&gt;
+   &lt;description&gt;
+    &lt;p&gt;
+     Use SweetHome 3D to map out a plan of your house and plan the layout
+     and colors of floors, walls and furniture.
+    &lt;/p&gt;
+    &lt;p&gt;Example list:&lt;/p&gt;
+    &lt;ul&gt;
+     &lt;li&gt;First item&lt;/li&gt;
+     &lt;li&gt;Second item&lt;/li&gt;
+    &lt;/ul&gt;
+    &lt;p&gt;
+    This is another paragraph of information about SweetHome 3D and the Many
+    things that you can use it for
+    &lt;/p&gt;
+   &lt;/description&gt;
+   &lt;screenshots&gt;
+    &lt;screenshot type="default"&gt;
+     &lt;image&gt;http://www.sweethome3d.com/images/gallery/SweetHome3DExample10-FlatWithMezzanine-VirtualVisit.jpg&lt;/image&gt;
+     &lt;caption&gt;The main window showing the application in action&lt;/caption&gt;
+    &lt;/screenshot&gt;
+   &lt;/screenshots&gt;
+   &lt;url type="homepage"&gt;http://http://www.sweethome3d.com/&lt;/url&gt;
+  &lt;/component&gt;
+  </pre>
+
+  The sweethome 3D tarball also does not ship with an icon included, so we need to download it from their upstream source control:
+
+  <pre>
+  <span class="unselectable">$ </span>wget http://sweethome3d.cvs.sourceforge.net/viewvc/sweethome3d/SweetHome3D/src/com/eteks/sweethome3d/viewcontroller/resources/help/images/applicationIcon.png
+  </pre>
+
+  ### Place them in the sandbox
+  Now you have the 3 files in your working directory, we can place them in the appropriate locations in the build sandbox.
+
+  First, create the directory for the desktop file, and move the file from your working directory, to the build sandbox:
+
+  <pre>
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak mkdir -p /app/share/applications/
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak cp com.sweethome3d.App.desktop /app/share/applications/
+  </pre>
+
+  Then do the same for the icon, being sure to rename it to ```com.sweethome3d.App.png```
+  <pre>
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak mkdir -p /app/share/icons/hicolor/128x128/apps/
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak cp applicationIcon.png /app/share/icons/hicolor/128x128/apps/com.sweethome3d.App.png
+  </pre>
+
+  Finally, create the ```/app/share/appdata/``` directory in the sandbox, and copy the appdata file over:
+  <pre>
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak mkdir -p /app/share/appdata/
+  <span class="unselectable">$ </span>flatpak build sweethome3d-flatpak cp com.sweethome3d.App.appdata.xml /app/share/appdata/
+  </pre>
```

Comment:
This is a great start, although its missing a last step, which admittedly is not obvious or well documented.
After installing the desktop file and the appdata file you need to run something like
```
flatpak build sweethome3d-flatpak appstream-compose --prefix=/app --origin=flatpak --basename=com.sweethome3d.App  com.sweethome3d.App
```
This will read the desktop file, the icons and the appdata file and combine these details into an appstream files in `/app/share/app-info/`. These files are what flatpak extracts from every app in a repo and puts into the appstream branch.

For an example you can do:
```
flatpak update --appstream gnome-nightly-apps
```
Then look into `/var/lib/flatpak/appstream/gnome-nightly-apps/x86_64/active` to see how this looks.
This is the data that gnome-software uses for a remote.